src/utils/openCageApiUtils.ts

```diff
@@ -0,0 +1,43 @@
+import { Address, Geocode } from '../models/data_models/types';
+import opencage from 'opencage-api-client';
+import { DataSourceConfiguration } from '../models/data_models/dataSourceConfiguration';
+
+export class OpenCageApiUtils {
+	private apiKey: string = '';
+
+	constructor(private config: DataSourceConfiguration) {}
+
+	public async init(): Promise<this> {
+		this.apiKey = this.config.openCage.apiKey;
+		return this;
+	}
+
+	private reformatGeocodeResult(response: any): Address {
+		const { components, formatted, geometry } = response;
+		return {
+			formattedAddress: formatted,
+			city: components?.city,
+			stateCode: components?.state_code,
+			postalCode: components?.postcode,
+			countryCode: components?.country_code,
+			geocode: geometry,
+		};
+	}
+
+	public async addressLookup(address: string): Promise<Address | undefined> {
+		const results = await opencage.geocode({ q: address, key: this.apiKey }).then((data) => data?.results);
+
+		const result = results?.[0];
+
+		return this.reformatGeocodeResult(result);
```

Comment:
This oneliner might have better performance since we don't have to await/return

```suggestion
		return opencage.geocode({ q: address, key: this.apiKey })
			.then((data) => this.reformatGeocodeResult(data?.results.[0]));
```